src/Symfony/Bundle/FrameworkBundle/Command/CacheWarmupCommand.php

```diff
@@ -54,18 +91,24 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        // BC to be removed in 4.0
+        if (null === $this->cacheWarmer) {
+            $this->cacheWarmer = parent::getContainer()->get('cache_warmer');
+        }
+        if (null === $this->cacheDir) {
+            $this->cacheDir = parent::getContainer()->getParameter('kernel.cache_dir');
+        }
+
         $io = new SymfonyStyle($input, $output);
 
-        $kernel = $this->getContainer()->get('kernel');
+        $kernel = $this->getApplication()->getKernel();
         $io->comment(sprintf('Warming up the cache for the <info>%s</info> environment with debug <info>%s</info>', $kernel->getEnvironment(), var_export($kernel->isDebug(), true)));
 
-        $warmer = $this->getContainer()->get('cache_warmer');
-
         if (!$input->getOption('no-optional-warmers')) {
-            $warmer->enableOptionalWarmers();
+            $this->cacheWarmer->enableOptionalWarmers();
         }
 
-        $warmer->warmUp($this->getContainer()->getParameter('kernel.cache_dir'));
+        $this->cacheWarmer->warmUp(null === $this->cacheDir ? $kernel->getCacheDir() : $this->cacheDir);
```

Comment:
Nope. Above is for BC, as long as the command extends ContainerAwareCommand we need to get the cachedir from the container, if not provided already.

This is the new behavior if cachedir is not provided in 4.0.

However i still have my concerns about getCacheDir() vs getParameter('cachedir'). I think we should actually prefer getParameter(), also in 4.0.